app/models/use_cases/get_diagnoses.rb

```diff
@@ -4,16 +4,31 @@ module UseCases
   class GetDiagnoses
     class << self
       def for_user(user)
-        in_progress_associations = [visit: [facility: [:company]]]
-        completed_associations = [
-          visit: [:visitee, facility: [:company]], diagnosed_needs: [:selected_assistance_experts]
-        ]
         user_diagnoses = Diagnosis.of_user(user).reverse_chronological
+        in_progress_associations = [visit: [facility: [:company]]]
         {
           in_progress: user_diagnoses.in_progress.includes(in_progress_associations),
-          completed: user_diagnoses.completed.includes(completed_associations)
+          completed: completed_diagnoses_from_user_diagnoses(user_diagnoses)
         }
       end
+
```

Comment:
Could we imagine creating a query category of objects ? Like use cases but not really ?